File: CS-Examples/06_Comments/AddCommentWithAuthor.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace AddCommentWithAuthor
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            // Create a workbook
			Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\WorksheetSample1.xlsx");

            // Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Get the range in which the comment will be added (cell C1).
            CellRange range = sheet.Range["C1"];

            //Set the author and comment content
            string author = "E-iceblue";
            string text = "This is demo to show how to add a comment with editable Author property.";

            // Add comment to the range and set properties
            ExcelComment comment = range.AddComment();
            comment.Width = 200;
            comment.Visible = true;
            comment.Text = string.IsNullOrEmpty(author) ? text : author + ":\n" + text;

            // Set the font of the author
            ExcelFont font = range.Worksheet.Workbook.CreateFont();
            font.FontName = "Tahoma";
            font.KnownColor = ExcelColors.Black;
            font.IsBold = true;
            comment.RichText.SetFont(0, author.Length, font);

            // Specify the resulting file name.
            string output = "AddCommentWithAuthor.xlsx";

            // Save the modified workbook to a file using Excel 2013 format.
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the Excel file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: CS-Examples/06_Comments/AddCommentWithAuthor.vb
Imports Spire.Xls

Namespace AddCommentWithAuthor

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Instantiate a new Workbook object.
            Dim workbook As New Workbook()

            'Load an Excel document from the specified file.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\WorksheetSample1.xlsx")

            'Access the first worksheet in the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Get the CellRange object for cell C1.
            Dim range As CellRange = sheet.Range("C1")

            'Specify the author and text for the comment.
            Dim author As String = "E-iceblue"
            Dim text As String = "This is demo to show how to add a comment with editable Author property."

            'Add a comment to the range and customize its properties.
            Dim comment As ExcelComment = range.AddComment()
            comment.Width = 200
            comment.Visible = True
            comment.Text = If(String.IsNullOrEmpty(author), text, author & ":" & vbLf & text)

            'Create a new font and apply it to the author portion of the comment's rich text.
            Dim font As ExcelFont = range.Worksheet.Workbook.CreateFont()
            font.FontName = "Tahoma"
            font.KnownColor = ExcelColors.Black
            font.IsBold = True
            comment.RichText.SetFont(0, author.Length, font)

            'Specify the filename to save the modified workbook.
            Dim output As String = "AddCommentWithAuthor.xlsx"

            'Save the workbook to the specified file using Excel 2013 format.
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
